docs/usage/self-hosted-configuration.md

```diff
@@ -548,6 +548,20 @@ Information provided mainly in debug log level.
 Default execution timeout in minutes for child processes Renovate creates.
 If this option is not set, Renovate will fallback to 15 minutes.
 
+## experimentalFlags
+
+The `experimentalFlags` configuration option helps you manage experimental settings within Renovate.
+
+```javascript
+module.exports = {
+  experimentalFlags: ['dockerMaxPages=10'],
```

Comment:
```suggestion
  experimentalFlags: {
    dockerMaxPages: 10
  },
```
Is there a reason why we have a different way for defining experimentalFlags? 

From a users perspective this would add additional friction.

Reply:
Rhys suggested to keep it the current way: https://github.com/renovatebot/renovate/issues/27879#issuecomment-2002943040

Reply:
I would prefer to keep this consistent with the rest of the configuration system. IMO this would be a better UX and we can reuse more code.

@rarkins @viceice WDYT?

Reply:
We should be consistent. It's difficult enough to configure Renovate as is. Having exceptions is harder to think about and to maintain.

Reply:
I don't like this suggestion though. Would prefer to put this feature on hold if we don't have consensus to make it an array of strings

Reply:
I like the object style because we don't need to parse the strings 🤷‍♂️

Reply:
How many are non Boolean?

Reply:
4 with values and only one boolean

Reply:
In total there are 12 non boolean, 4 of which are converted in this PR.

Reply:
Can you list them?

Reply:
|name|type|
| --------------------------------------------------- | ----------------------------------------- |
| `RENOVATE_CACHE_NPM_MINUTES`                        | integer                                   |
| `RENOVATE_USER_AGENT`                               | string                                    |
| `RENOVATE_X_AUTODISCOVER_REPO_ORDER`                | string, [asc, desc]                       |
| `RENOVATE_X_AUTODISCOVER_REPO_SORT`                 | string, [alpha, created, updated,size,id] |
| `RENOVATE_X_DOCKER_MAX_PAGES`                       | integer                                   |
| `RENOVATE_X_GITLAB_AUTO_MERGEABLE_CHECK_ATTEMPS`    | integer                                   |
| `RENOVATE_X_GITLAB_BRANCH_STATUS_DELAY`             | integer                                   |
| `RENOVATE_X_GITLAB_MERGE_REQUEST_DELAY`             | integer                                   |
| `RENOVATE_X_MERGE_CONFIDENCE_API_BASE_URL`          | string                                    |
| `RENOVATE_X_MERGE_CONFIDENCE_SUPPORTED_DATASOURCES` | string array                              |
| `RENOVATE_X_PLATFORM_VERSION`                       | string                                    |
| `RENOVATE_X_S3_ENDPOINT`                            | string                                    |

Reply:
@nabeelsaabna how many of these do you think should be converted to real options now for your needs?

Reply:
@viceice how about the Gitea autodiscover ones? We could either:
- Promote to regular config options, or
- Always use a default value, have an experimental flag to swap it (would not support 5 options for repo sort but are they really necessary?)

Reply:
@viceice I think you also added `RENOVATE_X_PLATFORM_VERSION` for GitLab about 3 years ago and @RahulGautamSingh added or modified it for Bitbucket Server recently. Is the only point to "save an API call"? I would prefer either:
- Drop this capability, or
- If the only purpose is to say "we're on a latest enough version" then let that be a flag, or
- After 3 years it shouldn't be experimental?

Reply:
> @viceice how about the Gitea autodiscover ones? We could either:
> 
> * Promote to regular config options, or
> * Always use a default value, have an experimental flag to swap it (would not support 5 options for repo sort but are they really necessary?)

I think we can promote them to regular options.



> @viceice I think you also added `RENOVATE_X_PLATFORM_VERSION` for GitLab about 3 years ago and @RahulGautamSingh added or modified it for Bitbucket Server recently. Is the only point to "save an API call"? I would prefer either:
> 
> * Drop this capability, or
> * If the only purpose is to say "we're on a latest enough version" then let that be a flag, or
> * After 3 years it shouldn't be experimental?

We should make those to regular options too.

Reply:
So what if we change this PR to only migrate boolean options (where the presence of the flag means "true") so that it's a simple list of strings? I want this to be very easy to configure in env and/or cli

Reply:
🤷‍♂️

Reply:
OK. @RahulGautamSingh please simplify this by having boolean flags only (and no =false, if any exist)